### ISSUE TITLE
feature: specify remote shell / login command

### ISSUE THREAD

**Issue by thorstenhirsch** [icetee/remote-ftp#495]
Tools like WinSCP offer an option to specify the remote shell being used. This can also be used to configure an initial login command like "sudo su - someuser". Why not connecting as someuser in the first place? Well, we've got enterprise environments where the login user differs from the user that is allowed to work on files. So everybody needs to sudo su - user before he can start with his actual work. Putty allows to specify such a command, too.

**Comment by jimmaaay** [comment]
So in short are you looking to have something that would allow the execution of a command everytime you connect to the server?

**Comment by thorstenhirsch** [comment]
Yes. I wanted to give some context, so that you know that this command can
involve switching the user. Thus file synchronization might not be trivial.
I don't know.

Am 11.10.2016 5:29 nachm. schrieb "jimmaaay" notifications@github.com:

**Comment by jimmaaay** [comment]
As long as the server supports ssh commands should be doable

**Comment by ityreh** [comment]
Any news on this request? I have the same problem here, switching the user by `sudo su - techuser` if I want to edit some files outside of my home. If I could tell **remote-ftp** to execute this command right after the login it would be great! :+1:

**Comment by grenierdev** [comment]
Hi guys, been a while since I've posted in here.

I've been working on a project that has it's roots in remote-ftp : file system abstraction. A client of mine had needs to access SFTP under `sudo` like MobaXterm using SCP protocol (which support SUDO).

[SFTP protocol does not support any method to achieve this.](https://github.com/mscdex/ssh2/issues/626) However, you can emulate SFTP command with SSH command prefixed with `sudo -s` or `sudo -i -u jimmy`, like [I did here](https://github.com/konstellio/fs/blob/master/src/FileSystems/SSH2.ts#L180-L187).
